### PR TITLE
Upgrade react-intl

### LINKED ISSUE #22760: In Google Chrome, times between midnight and 1AM are shown as, e.g. 24:17 instead of 00:17 due to an issue with react-intl
### Steps to reproduce the problem

1. Create a post between midnight and 1 AM local time
2. Navigate to the post using Google Chrome

### Expected behaviour

Timestamp is displayed as 00:17 if the post was made at 12:17AM

### Actual behaviour

Timestamp is displayed as 24:17 if the psot was made at 12:17AM

### Detailed description

react-intl did not support the hourCycle='h23' option until later versions. We are still using react-intl v2.9 which is very, very old and includes breaking changes to migrate to v3.

We should replace usages of react-intl with direct browser API calls, which have stabilized since this issue came up and are well supported across browsers.

### Specifications

Mastodon 4.0.2
react-intl v2.9.0

### PR DESCRIPTION
**Pull request by renchap**
See the various commits for details of what is does.

TODO:
- [x] Ensure that the Crowdin process still works
- [x] Replace `manage:translations`
- [x] Ensure polyfills are loaded correctly (https://formatjs.io/docs/polyfills)
- [x] See if we can replace `<RelativeTimestamp>` with something simpler based on `react-intl` components or helpers, but I am not sure what all the props do. **status**: should be in a future PR, not really needed right now
- [x] Correctly handle `oc` and `sa` locales (see below)
- [x] Ensure no errors occur when we try to load polyfills for a non-existing locale (`oc`). **status**: `shoudPolyfillPluralRules` has a built-in list of supported locales, and fallbacks to `en`, so all here.

## Custom plural rules issue

Currently we are using `sa` and `oc` as custom plural rules, but they are not supported by the Javascript Intl API, resulting is this message on page load:
`formatjs/intl Error MISSING_DATA] Missing locale data for locale: "oc" in Intl.NumberFormat. Using default locale: "en" as fallback. See https://formatjs.io/docs/react-intl#runtime-requirements for more details`

I am not sure how to properly handle this.

The browser is supposed to pick the best fit for the provided locale, so I guess we could just make FormatJS not print this message (using `onError`) for those 2 locales?

### ✅ Solution

`oc` is not supported in Chrome & Firefox, `sa` is not supported in Chrome, both are supported in Safari. According to the JS spec, the browsers should pick the best available locale here, so we can simply silence the message.

## Fixes

- #22760

### CODE REVIEW

**Comment by renchap** [comment]
@Quenty31 I see you contributed to the Occitan translation before, do you have an opinion on the custom plural issue I described above?

**Comment by renchap** [comment]
Added polyfills for `Intl`. See the commit message for details.

**Comment by brawaru** [comment]
Doesn't seem that `oc` (Occitan) and `sa` (Sanskrit) languages have plural rules defined:
- https://github.com/unicode-org/cldr-json/tree/main/cldr-json/cldr-numbers-full/main/oc
- https://github.com/unicode-org/cldr-json/tree/main/cldr-json/cldr-numbers-full/main/sa

They maybe can contribute the rules to CLDR, otherwise this warning about locale data is probably OK to ignore. It shouldn't cause any issues and Intl should resort to using default locale rules.

**Comment by renchap** [comment]
After more testing:

- Safari supports `oc` and `sa` locales
- Firefox only supports `sa` locale
- Chrome does not support `sa` and `oc` at all
- 
I think we are fine with silencing the error in production, and relying on the browser to pick the best relevant supported locale.

**Comment by renchap** [comment]
Nearly done! The last commits handle the non-supported locales as explained in the initial post.

`yarn manage:translations` has been removed as it is no longer working and very outdated, we should be able to work without it using the new formatjs CLI. Documentation needs to be updated after the change.

The last remaining issue is that extraction is configured to write into `en.json`, but the default messages do not match the current english translation, so it overrides them. Maybe keys & default messages needs to be extracted to `defaultMessages.json` instead?

**Comment by renchap** [comment]
One more try:
- removed `defaultMessages.js` as it is not currently used anywhere (CrowdIn ingests the `en.json` file)
- matched current behaviour for I18n keys extraction: if a key exists in `en.json`, then we use the translation defined in this file, not the extracted `defaultMessage`

Running `yarn i18n:extract` with my latest commit does not change `en.json` at all, which seems correct. Adding a new key in a JS file correctly adds the key to `en.json`.

**Comment by renchap** [comment]
`formatjs extract` (correctly) reported that some keys were defined with different `defaultMessage` at different places in the codebase.

This is now fixed in the last commit, and I changed the GH:A workflow so it will fail if `formatjs extract` detect any issues.

This PR should be now ready for review!

**Comment by ClearlyClaire** [comment]
It appears `app/views/layouts/embedded.html.haml` still needs to get updated.

I'd also possibly move e219447bdab51a0e1d866f4b036675d4ff244fe2 to another PR

**Comment by renchap** [comment]
> It appears `app/views/layouts/embedded.html.haml` still needs to get updated.

Done

> I'd also possibly move [e219447](https://github.com/mastodon/mastodon/commit/e219447bdab51a0e1d866f4b036675d4ff244fe2) to another PR

#25114

File: app/javascript/mastodon/polyfills/intl.ts
Comment (ThisIsMissEm) [inline app/javascript/mastodon/polyfills/intl.ts:1]: @renchap does this file come from somewhere? Kinda curious about all the commented code here?

**Review by ThisIsMissEm — COMMENTED** [review]
I've reviewed this code, and it looks good to me, though I might not be aware of everything that this affects. Just one or two minor comments.

File: app/javascript/mastodon/polyfills/intl.ts
Comment (renchap) [reply app/javascript/mastodon/polyfills/intl.ts:1]: No, I wrote it, but as explain in the relevant commit (I think) I wanted to let those here in case we notice they need to be enabled because we have user reporting it does not work in a given (old) browser.

I spent some time writing this and I did not wanted to redo it if it was needed :)

**Review by ClearlyClaire — APPROVED** [review]
It will likely be a nightmare for me to deal with in glitch-soc, but it's a welcome modernization, `injectIntl` has made things more complicated more than once.

The PR looks fine to me, but since it's a large-scale change, I'm waiting for a ack from @Gargron before merging.

**Review by Gargron — APPROVED** [review]

**Comment by ClearlyClaire** [comment]
Note that this breaks cross-origin loading of the assets because of the missing `crossorigin` attribute on the preload tags.

**Pull request merged**
